✨ 버스정류장 리스트 추가

hyeonseo1109 opened the pull request:
open api 이용하여 버스정류장 리스트 추가함.

hyeonseo1109 committed:
✨ 버스정류장 리스트 추가

open api 이용하여 버스정류장 리스트를 추가하였음.

hyeonseo1109 committed:
💄 정렬 변경

좌측정렬로 변경

hyeonseo1109 committed:
임시저장

hyeonseo1109 committed:
✨ 실시간 버스 위치 표시

open api 활용하여 실시간 버스 위치를 표시함.

hyeonseo1109 committed:
💄 Css 변경

좌측에 line을 넣어 버스 아이콘이 line을 따라 움직이게 하였음.

hyeonseo1109 committed:
💄 Css 변경

좌측 line 색상 변경

hyeonseo1109 committed:
🐜 fix : 오류  수정

실시간 버스 정보가 없거나 1대만 있을 경우,
받아오는 값이 배열이 아니게 되므로 오류가 발생하게 됨.
function BusLocation의 setLocation으로 변수 busLocations가 무조건 배열이 되도록 함.

hyeonseo1109 committed:
💄 Css 변경 (line)

좌측 line에 정류소마다 원 추가, 회차지 하단 hr 위치 수정

hyeonseo1109 committed:
오류 확인

the-eris-net reviewed (commented):
따봉

the-eris-net commented on src/busStation.jsx:
```diff
@@ -0,0 +1,45 @@
+    useEffect(() => {
+        const getData = async () => {
+            try {
+                const data = await fetch(`https://apis.data.go.kr/6410000/busrouteservice/v2/getBusRouteStationListv2?serviceKey=GS5aLVDaFTPnk4vYpexWNPYAl22LeqGp3N5duJZ3OqXXCfMOz0%2FleRDyv%2B6xHXvIogUG8jV2gCGaG04Hs39uRQ%3D%3D&routeId=${id}&format=json`);
```
P4 : url를 코드에 직접 넣는것은 좋지 않습니다.
```javascript
const path = "https://apis.data.go.kr "
const url = "6410000/busrouteservice/v2/getBusRouteStationListv2"
```
등으로 분리할수 있습니다. 
코드에 직접 문자열을 넣는것은 보통은 권장되지 않고 변수로 빼는것이 좋습니다

변수로 뺄 경우 파일로 따로두거나 하는 방법도 있습니다

the-eris-net commented on src/busStation.jsx:
```diff
@@ -0,0 +1,45 @@
+                <div className='bar'></div>
+                
+                <ul className='busStopList'>
+                    {stations.map((station) => (
```
👍👍👍👍
map을 사용해서 component를 출력하는게 처음 하면 어려울 수도 있는데 잘했습니다

the-eris-net commented on src/busStation.jsx:
```diff
@@ -0,0 +1,45 @@
+                <ul className='busStopList'>
+                    {stations.map((station) => (
+                        <>
+                            <li className="busStop" key={`${station.stationId}-${station.stationSeq}`}>
```
👍👍👍
key를 유니크한걸 잘 선택하신거 같습니다.
3따봉 적립

the-eris-net commented on src/current.jsx:
```diff
@@ -0,0 +1,39 @@
+        };
+        getData();
+
+        const interval = setInterval( () => {
```
interval 대표적인 메모리 누수가 발생할수 있는 부분입니다
그래서 항상 지워주는 방식으로 코드를 작성해야합니다

the-eris-net commented on src/current.jsx:
```diff
@@ -0,0 +1,39 @@
+                // 실시간 버스가 여러 대 있을 때는 busLocationList가 배열로 오지만, 
+                // 딱 1대만 있을 때는 단일 객체 형태로 오고, 아예 없으면 null로 올 수 있음. 
+                console.log('real-time locations:', busLocations);
+                setLocation(busLocations ? (Array.isArray(busLocations) ? busLocations : [busLocations]) : []);
```
P5 : 중첩 3항연산자는 코드 가독성에 좋지 않습니다.
만약 중첩될거 같으면 함수를 빼던가 아니면 들여쓰기를 하는게 좋지 않을까요?

the-eris-net commented on src/busStation.jsx:
```diff
@@ -0,0 +1,45 @@
+                            <li className="busStop" key={`${station.stationId}-${station.stationSeq}`}>
+                                <span className='circle'></span>
+                                {station.stationName}
+                                {locations.some(location => Number(location.stationId) === Number(station.stationId)) && (<span className='icon'>🚌</span>)}
```
location.stationId랑 station.stationId은 둘다 숫자로 들어올것 같은데 왜 Number로 감싼건가요?🤔🤔🤔🤔🤔🤔

hyeonseo1109 committed:
🎨 코드 개선

중첩되는 3항연산자 -> if문으로 변

hyeonseo1109 committed:
🎨 코드 변경

코드 내의 url을 변수로 빼 내었음.

hyeonseo1109 committed:
🔥 필요 없는 코드 삭제

기존 정류장의 아이디와 현재위치의 아이디를 비교할 때 Number 씌웠던 것 삭제함

hyeonseo1109 committed:
💄 Css 변경

컴퓨터 환경과 달리 모바일 환경에서는 bar와 icon&circle의 배치가 어긋나는 오류 발견, 이에 위치설정 단위를 px -> em 으로 변경하였음.

hyeonseo1109 committed:
💄 Title 하단에 구분선 추가하였음.

Title 하단에 구분선 추가하였음.

the-eris-net commented on src/busStation.jsx:
```diff
@@ -0,0 +1,48 @@
+    const [stations, setStations] = useState([]);
+    const id = 219000013;
+    const path = "https://apis.data.go.kr";
+    const url = "6410000/busrouteservice/v2/getBusRouteStationListv2?serviceKey=GS5aLVDaFTPnk4vYpexWNPYAl22LeqGp3N5duJZ3OqXXCfMOz0%2FleRDyv%2B6xHXvIogUG8jV2gCGaG04Hs39uRQ%3D%3D&"
```
key는 숨기는게 좋습니다.
보통 변수로 넣고 .env, 환경변수로 넣습니다
